atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -571,6 +575,25 @@ private Iterator<Map.Entry<Cell, byte[]>> getSortedColumnsLocalWrites(
                 cellComparator);
     }
 
+    @VisibleForTesting
+    static void reportExpectationsCollectedData(
+            ExpectationsAwareTransaction transaction, ExpectationsMetrics metrics, boolean isStillRunning) {
+        if (isStillRunning) {
+            log.error("reportExpectationsCollectedData is called on an in-progress transaction");
```

Comment:
The only meh thing here is that we won't log `state` here. I had added it in the old old PR because this code path was hit. Now that that's fixed, it's not needed (we could always RC if it comes up) as the conditional logic was fixed!